src/App.js

```diff
@@ -0,0 +1,113 @@
+        super()
+        this.state = {
+            newToDoItem: '',
+            toDoItems: [],
+            persistData: false,
+            incorrectEntry: false
+        };
+        this.localStorageItems = 'items';
+    }
+
+
+
+    componentDidMount() {
+        const storageItems = localStorage.getItem(this.localStorageItems);
+        let parsedItems;
+        if (storageItems) {
+            try {
+                parsedItems = JSON.parse(storageItems);
+            } catch (e) {
+                alert(e);
+            }
+        }
+        if (parsedItems) {
+            const items = parsedItems.reduce((arrayItems, item) => {
+                arrayItems.push({
+                    value: item.value,
+                    isChecked: item.isChecked
+                });
+                return arrayItems;
+            }, []);
+
+            this.setState({ toDoItems: items });
+        }
+    }
+
+    componentDidUpdate() {
+        if (this.state.persistData) {
+            this.saveToLocalStorage();
+        }
+    }
+
+    onInputChange = event => {
+        this.setState({ newToDoItem: event.target.value })
+    }
+
+    onAddButtonClick = (e) => {
+        e.preventDefault();
+        if (this.state.newToDoItem) {
+            this.setState({
+                toDoItems: this.state.toDoItems.concat({
+                    value: this.state.newToDoItem,
+                    isChecked: false
+                }), newToDoItem: '', incorrectEntry: false
+            })
+        } else {
+            this.setState({ incorrectEntry: true });
+        }
+    }
+
+    onDeleteButtonClick = item => {
+        this.setState({
+            toDoItems: this.state.toDoItems.filter(function (singleItem) {
+                if (singleItem.value !== item.value) {
+                    return singleItem;
+                }
+            })
+        });
+    }
+
+    saveButtonToggleClass = () => {
+        const currentState = this.state.persistData;
+        this.setState({ persistData: !currentState });
+    }
+
+    itemToggleClass = item => {
+        const items = this.state.toDoItems.map((input) => {
+            return item === input ? { ...item, isChecked: !item.isChecked } : input;
+        })
+
+        this.setState({ toDoItems: items });
+    }
+
+    saveToLocalStorage = () => {
+        try {
+            localStorage.setItem(this.localStorageItems, JSON.stringify(this.state.toDoItems));
+        } catch (e) {
+            alert(e);
+        }
+    }
+
+    removeFromLocalStorage = () => {
+        localStorage.removeItem(this.localStorageItems);
+    }
+
+    render() {
+        return (
+            <div className='container'>
+                <AppHeader inputClass={this.state.incorrectEntry ? 'input input--incorrect' : 'input'} inputChange={this.onInputChange} addItem={this.onAddButtonClick} inputValue={this.state.newToDoItem} errorMessageClass={this.state.incorrectEntry ? 'error-message error-message--visible' : 'error-message'}/>
```

Comment:
ok, now as you've cleaned up a little bit of logic, we can go to restructuring.

first thing, can you please create a new directory `modules` and inside of it `todo-list` directory?
after that can you please move out all logic from the app into a new component (you can name it like `TodoList`) inside of `modules/todo-list` directory?

next thing, inside of this component lift up logic from `AppHeader` and `AppFooter` to the new `TodoList` (this would lead to removal of `AppHeader` and `AppFooter`)

next thing, please add to the project react-router (https://reactrouter.com/) and please in App component add routing for `/` path which would render the new `TodoList` component

Reply:
I'm not sure if I understand you well. Should I move everything from `App.js` into the new `ToDoList` component? So should I remove the `App.js`?

Reply:
Ok, I changed the structure and added the react-router.